onnxruntime/contrib_ops/cpu/quantization/qlinear_softmax.cc

```diff
@@ -0,0 +1,373 @@
+          const uint32_t* shifted_lookuptable = lookup_table;
+          size_t elements_n = D;
+          // reduceSumUin8ToUint32: need speedup
+          uint32_t vsum = 0;
+          const int8_t* x_t_cur = x_t;
+          do {
+            const size_t vx = uint8_t(adjustment + *x_t_cur++);
+            vsum += shifted_lookuptable[vx];
+          } while (--elements_n != 0);
+          if (vsum == 0) {
+            return;
+          }
+          elements_n = D;
+          x_t_cur = x_t;
+          // elementwise div
+          const uint32_t vrounding = (vsum >> 1);
+          do {
+            const size_t vx = uint8_t(adjustment + *x_t_cur++);
+            const uint32_t vt = shifted_lookuptable[vx];
+            // simulate round function, and re-quant to int8
+            const uint32_t vq = ((vt * c_y_scale) + vrounding) / vsum + c_y_zp;
+            const int8_t vy = static_cast<int32_t>(vq) > 255 ? static_cast<int8_t>(255) : static_cast<int8_t>(vq);
+            *y_t++ = vy;
+          } while (--elements_n != 0);
+          x_t = x_t_cur;
+        }
+      });
+
+  return Status::OK();
+}
+
+template <typename T>
+const uint32_t* QLinearSoftmax<T>::GetLookupTable(OpKernelContext* context, size_t reduce_len) const {
+  const uint32_t* lookup_table = fixed_lookup_table_.data();
+  if (fixed_lookup_table_.size() == 0) {
+    tmp_lookup_table_.resize(256);
+    lookup_table = tmp_lookup_table_.data();
+    const float X_scale = *(context->Input<Tensor>(1)->Data<float>());
+    QlinearBuildLookupTableUint32<T>(tmp_lookup_table_.data(), X_scale, reduce_len);
+  }
+  return lookup_table;
+}
+
+// opset-12 and below
+template <typename T>
+Status QLinearSoftmax<T>::ComputeImpl(OpKernelContext* context, const Tensor& input, Tensor& output,
+                                      concurrency::ThreadPool* thread_pool,
+                                      const uint32_t* lookup_table) const {
+  const auto* Y_scale_tensor = context->Input<Tensor>(3);
+  const auto* Y_zp_tensor = context->Input<Tensor>(4);
+  const uint32_t Y_scale = gsl::narrow_cast<uint32_t>(1.0f / (*(Y_scale_tensor->Data<float>())));
+  const T Y_zp = Y_zp_tensor ? *(Y_zp_tensor->Data<T>()) : 0;
+
+  const auto& X_shape = input.Shape();
+  const size_t N = X_shape.SizeToDimension(axis_);
+  const size_t D = X_shape.SizeFromDimension(axis_);
+  return QlinearSoftmaxCPU<T>(N, D, input.template Data<T>(), output.template MutableData<T>(),
+                              lookup_table, Y_scale, Y_zp, thread_pool);
+}
+
+// opset-13 and above
+template <typename T>
+Status QLinearSoftmax<T>::ComputeImplOpset13(OpKernelContext* context,
+                                             const Tensor& input, Tensor& output,
+                                             concurrency::ThreadPool* thread_pool,
+                                             const uint32_t* lookup_table) const {
+  const auto* Y_scale_tensor = context->Input<Tensor>(3);
```

Comment:
After refactored of removing templatized code, this file is accounting for `26.1Ki`, it should be not too much.

After consideration, I choose to keep `QlinearSoftmaxCPU` even if this two function may take a lot VMsize, keeping the two function clear and faster.  And it may require speedup with some intrinsic instructions in different platform afterwards.

Reply:
On-disk size is more important. There's a size limit for mobile apps in the Google Play store that some partners are very close to.

Reply:
Vmsize and filesize has the equal value, `22.2ki` after stripped debuginfos.